### PR TITLE
[DataFrame] Implements df.duplicated and df.drop_duplicates

### PR DESCRIPTION
**Pull request by SaladRaider**
## What do these changes do?
Implements df.duplicated and df.drop_duplicates

### CODE REVIEW

**Review by kunalgosar — CHANGES_REQUESTED** [review]
Took a quick cursory look through, and have some high level comments. Need to come back and take a deeper look.

File: python/ray/dataframe/dataframe.py [hunk 1]
```diff
@@ -1503,14 +1503,127 @@ def drop_helper(obj, axis, label):
...
+        Returns:
+            duplicated: Series
+        """
+        pd.DataFrame(columns=self.columns).duplicated(subset=subset, keep=keep)
+        if subset is None:
+            subset = self.columns
+        elif (not np.iterable(subset) or
+              isinstance(subset, compat.string_types) or
+              isinstance(subset, tuple) and subset in self.columns):
+            subset = subset,
```
Comment (Veryku) [inline python/ray/dataframe/dataframe.py]: Hanging comma

**Review by p-yang — COMMENTED** [review]
Could you describe the process for `duplicated`? It's a bit unclear right now.

File: python/ray/dataframe/dataframe.py (hunk 1 above)
Comment (SaladRaider) [reply python/ray/dataframe/dataframe.py]: this is to make the subset a tuple. do you think it would be better to write `subset = (subset,)`?

Comment (kunalgosar) [reply python/ray/dataframe/dataframe.py]: `(subset,)` is more idiomatic.

File: python/ray/dataframe/test/test_dataframe.py [hunk 2]
```diff
@@ -1254,17 +1254,132 @@ def test_drop_api_equivalence():
 
 
 def test_drop_duplicates():
```
[the remark this answers is not in the document]
Comment (SaladRaider) [reply python/ray/dataframe/test/test_dataframe.py]: done!

**Comment by YoannMR** [comment]
Hi,
Has drop_duplicates been implemented?
If so, how can I update my Ray install to reflect this?
Thanks a lot!

**Comment by kunalgosar** [comment]
Hi @YoannMR, this implementation of `drop_duplicates` has not been merged as it is not ready yet. Once it is merged, you will be able to build Ray from source on master; that will contain the `drop_duplicates` code.

**Comment by YoannMR** [comment]
Thanks @kunalgosar !

Do I need to get access rights to CI server?

**Comment by kunalgosar** [comment]
No, when opening a new Pull Request, it will automatically trigger the CI server to build and test your code.

If you mean to build Ray from source locally on your computer, you can do so using the instructions at: http://ray.readthedocs.io/en/latest/installation.html#building-ray-from-source.

**Comment by robertnishihara** [comment]
@kunalgosar is this PR still in progress?

**Comment by kunalgosar** [comment]
@robertnishihara The PR is not complete and not ready to be merged, but I think it is unlikely that it will be completed soon. I'm fine with closing it for now and I can revisit implementing this at a later time.

**Pull request closed without merging**